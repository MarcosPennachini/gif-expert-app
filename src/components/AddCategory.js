import React, {useState} from 'react';
import PropTypes from 'prop-types'

const AddCategory = ({categories, setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCategories( [...categories, inputValue] );
            setInputValue('');
        }
        
    }

    const handleChange = (e) => {
        setInputValue(e.target.value);
    }
    return ( 
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder="Agrega una categorÃ­a"
                value={inputValue}
                onChange={ handleChange }
            >
            </input>
        </form>
     );
}
 
export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
