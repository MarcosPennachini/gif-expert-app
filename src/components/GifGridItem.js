import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title, img}) => {

    console.log(id, title, img);
    return (
        <div className="card animate__animated animate__jackInTheBox animate__fast">
            <img src={img} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}


GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
}
