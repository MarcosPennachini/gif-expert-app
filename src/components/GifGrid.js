import React, {useEffect, useState} from 'react';
import { GifGridItem } from './GifGridItem';
import getGifs from '../helpers/getGifs';

const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        return getGifs(category)
            .then( imgs => setImages(imgs));
    }, [])

    

    return ( 
        <>
            <h3>{category}</h3>
            <div className="card-grid">
            
                {images.map( (img) => {
                    return <GifGridItem 
                        key={img.id}
                        {...img}
                    /> 
                } )}
            </div>
        </>
    );
}
 
export default GifGrid;