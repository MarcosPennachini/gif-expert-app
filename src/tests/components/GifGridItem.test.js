import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import { GifGridItem } from '../../components/GifGridItem'

const id = "id";
const title = "title";
const img = "url";

const wrapper = shallow(<GifGridItem id={id} title={title} img={img} />);

describe('Pruebas en GiftGridItem', () => {

    test('<GiftGridItem /> se debe renderizar correctamente', () => {

        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('<GifGridItem debe tener un párrafo con el título pasado por props />', () => {
        const p = wrapper.find('p');
        expect(p.text()).toBe(title)
    })

    test('<GifGridItem debe tener un <img> con src y alt = a img y title pasado por props', () => {
        const image = wrapper.find('img');
        expect(image.prop('src')).toBe(img);
        expect(image.prop('alt')).toBe(title);
       
    })
})