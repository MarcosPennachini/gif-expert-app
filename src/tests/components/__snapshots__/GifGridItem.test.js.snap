// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftGridItem <GiftGridItem /> se debe renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__jackInTheBox animate__fast"
>
  <img
    alt="title"
    src="url"
  />
  <p>
    title
  </p>
</div>
`;
